standalone/lib/makefile

```diff
@@ -1,24 +1,31 @@
 # lib makefile
-SHELL=/bin/sh
+SHELL = /bin/sh
 RANLIB = ranlib
 RM = /bin/rm
-CC= gcc-fsf-6
-COPTS= -O -Wall -Wno-char-subscripts -Wno-maybe-uninitialized -Werror
+CC = gcc
+LD = ld
+DCOPTS = -O2 -Wall -Wno-char-subscripts -Wno-uninitialized -Wno-unused-result -Werror -fPIC
```

Comment:
@kalmarek question: you write that you had to use `-Wno-uninitialized` in order to compile with clang. But I don't get any warning with clang (actually, clang is my default compiler, as I work on OS X). Could you clarify which warnings you get, with which clang version, on which OS?

Reply:
```bash
$ make CC="clang"                         
clang -c -O -Wall -Wno-char-subscripts -Wno-maybe-uninitialized -Werror fsa.c 
error: unknown warning option '-Wno-maybe-uninitialized'; did you mean '-Wno-uninitialized'? [-Werror,-Wunknown-warning-option]
make: *** [makefile:24: fsa.o] Error 1
on v1.5.6:
$ clang -v                                    
clang version 7.0.1 (tags/RELEASE_701/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-pc-linux-gnu/8.2.1
Found candidate GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/8.2.1
Found candidate GCC installation: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1
Found candidate GCC installation: /usr/lib64/gcc/x86_64-pc-linux-gnu/8.2.1
Selected GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/8.2.1
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Selected multilib: .;@m64
```
whereas:
```bash
$ make CC="gcc" COPTS="-O2 -Wall -Wno-char-subscripts -Werror"
gcc -c -O2 -Wall -Wno-char-subscripts -Werror fsa.c 
gcc -c -O2 -Wall -Wno-char-subscripts -Werror fsaio.c 
gcc -c -O2 -Wall -Wno-char-subscripts -Werror hash.c 
gcc -c -O2 -Wall -Wno-char-subscripts -Werror rwsreduce.c 
gcc -c -O2 -Wall -Wno-char-subscripts -Werror miscio.c 
gcc -c -O2 -Wall -Wno-char-subscripts -Werror rwsio.c 
rwsio.c: In function ‘print_kboutput’:
rwsio.c:728:21: error: ‘i1’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
         if (i == i1 + 1)
                  ~~~^~~
cc1: all warnings being treated as errors
make: *** [makefile:24: rwsio.o] Error 1
```
(clang compiles with `-O2/3` without problem`)